### ISSUE TITLE
Add nef to polygon soup

### ISSUE THREAD

**Issue by sloriot** [CGAL/cgal#1657]
Add a function to convert a Nef_3 to a polygon soup (compatible with pmp's functions). That way even if the nef is not a manifold surface, we can still export it. I guess we should also have a function to write a polygon soup to OFF (even if it is trivial).

I think it is rather trivial. Index vertices, then iterate over halffacets:
 -  if the facet is dangling (the volume mark on both side is identical) only output the half-facet which handle is the smaller
- otherwise, display the half-facet on the unbounded side.

A half-face is incident to a bounded volume iif `hf->incident_volume()->mark()==true`.

**Comment by kintel** [comment]
I'd love to see a robust implementation of this.

As Nef_3 faces can have holes, you'd need to tessellate before creating a polygon soup. I've referenced some issues related to tessellation here: http://cgal-discuss.949826.n4.nabble.com/Nef3-Wrong-normal-vector-reported-causes-triangulator-crash-tt4660282.html

I have an STL implementation of this which is pretty robust, but the code is not pretty at all:
https://github.com/openscad/openscad/blob/master/src/cgalutils.cc#L261

The above code still struggles with side-effects (topology changes caused by quantizing of vertices) of down-converting to the floating point representation needed by STL (which is also needed by OFF).

In short: I dispute that this would be "rather trivial", unless the purpose is merely visualization :)

**Comment by sloriot** [comment]
Almost fixed by #3561 (documentation needs to be added but an arbitrary container must first be used in the implementation instead of `std::vector`).

**Comment by sloriot** [comment]
Fixed by #3599, it will be part of CGAL 5.0.
